### PR TITLE
#162949003 Implement pagination

### PR DESCRIPTION
**Pull request by JoshuaKodhe**
### What does this PR do?

Have pagination support for articles

### Description of Task to be completed?

Have pagination functionality

### How should this be manually tested?
**Run the server:**
`python manage.py runserver`

**From postman:**
    1. Create multiple articles by sending  `post` requests to the '/api/v1/articles/' endpoint with the data body 

```source-json
    {
        "article": {
        "title": "Test Article",
        "description": "Test description for the article",
        "body": "Test body for the article"
        }
    }
```

2. Retrieve all articles by sending a `get` request to the  '/api/v1/articles/' endpoint
**Default return without page_size**
```source-json
{
    "article": {
        "count": 2,
        "next": "http://127.0.0.1:8000/api/v1/articles/?page=2",
        "previous": null,
        "results": [
            {
                "title": "How to train your dragon",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-1",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png"
                },
                "created_at": "2019-01-28T09:30:06.694979Z",
                "updated_at": "2019-01-28T09:30:06.695002Z"
            }
        ]
    }
}
```
**Retrieving articles from a certain page with the default page_size, 8 articles returned**
`get` request to the  '/api/v1/articles/?page=2' endpoint
```source-json
{
    "article": {
        "count": 18,
        "next": null,
        "previous": "http://127.0.0.1:8000/api/v1/articles/",
        "results": [
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-7",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:09.288375Z",
                "updated_at": "2019-01-29T14:53:09.288401Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-6",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:08.865539Z",
                "updated_at": "2019-01-29T14:53:08.865564Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-5",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:08.415634Z",
                "updated_at": "2019-01-29T14:53:08.415688Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-4",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:07.696695Z",
                "updated_at": "2019-01-29T14:53:07.696731Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-3",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:06.854602Z",
                "updated_at": "2019-01-29T14:53:06.854625Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-2",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:06.026290Z",
                "updated_at": "2019-01-29T14:53:06.026355Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version-1",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:53:04.913958Z",
                "updated_at": "2019-01-29T14:53:04.913981Z"
            },
            {
                "title": "How to train your dragon version",
                "description": "Ever wonder how?",
                "body": "It takes a Jacobian",
                "slug": "how-to-train-your-dragon-version",
                "avg_rating": {
                    "avg_rating": 0,
                    "total_user_rates": 0,
                    "each_rating": {}
                },
                "image_url": null,
                "author": {
                    "username": "joe",
                    "bio": "",
                    "image": "https://d1nhio0ox7pgb.cloudfront.net/_img/o_collection_png/green_dark_grey/512x512/plain/user.png",
                    "followers_count": 0,
                    "following_count": 0,
                    "created_at": "2019-01-29T14:51:36.690699Z",
                    "updated_at": "2019-01-29T14:51:36.690731Z"
                },
                "created_at": "2019-01-29T14:52:55.801734Z",
                "updated_at": "2019-01-29T14:52:55.801769Z"
            }
        ]
    }
}
```

### What are the relevant pivotal tracker stories?
[#162949003](https://www.pivotaltracker.com/story/show/162949003)

### Screenshots
**Default get request (with no parameters set)**
screenshot 2019-01-28 at 19 12 32

**get request of the second page and a page_size of five (five articles a page)**
screenshot 2019-01-30 at 11 54 38

### CODE REVIEW

**Review by Nicanor008 — COMMENTED** [review]
I like how you use standard pagination in a simple way.This can also be re-used in future for displaying authors

**Review by Hesbon5600 — COMMENTED** [review]
Tested the PR and it looks amazing

File: authors/settings.py [hunk 1]
```diff
@@ -190,7 +190,10 @@
     'DEFAULT_AUTHENTICATION_CLASSES': (
         'authors.apps.authentication.backends.JWTAuthentication',
     ),
+
 }
+PAGE_SIZE = os.getenv("PAGE_SIZE")
```
Comment (eduhmik) [inline authors/settings.py]: Hey @JoshuaKodhe great way of implementing pagination, allows us to quickly change page size from our environment variable

**Review by kzyangiro — CHANGES_REQUESTED** [review]
@JoshuaKodhe Page size should not be in the settings/environment. Set a default page size i.e 10, then let users to be able to give the **page** - i.e page 2 and **page size** - Number of items to display in the page. These two vars should be passed by a user as params

You can then add about 15 items and test out what displays in page 1 and page 2 given a particular set of parameters, and show us the screenshot

File: authors/apps/articles/tests/test_pagination.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    def test_article_pagination(self):
+        self.is_present('count')
+        self.is_present('next')
+        self.is_present('previous')
+
+    def test_pagination_more_than_ten_articles(self):
+        self.create_multiple_articles()
+        response = self.client.get(self.create_list_article_url)
+        self.assertEqual(21, response.data['count'])
```
Comment (Nyakaru) [inline authors/apps/articles/tests/test_pagination.py:21]: Good use of DRY code.

**Review by kzyangiro — APPROVED** [review]

**Pull request merged**